stk/calculators/optimization/macromodel.py

```diff
@@ -1042,6 +1110,96 @@ def __init__(self,
                          minimum_gradient=minimum_gradient,
                          use_cache=use_cache)
 
+    @staticmethod
+    def _check_params(
+        temperature,
+        conformers,
+        simulation_time,
+        time_step,
+        eq_time,
+        minimum_gradient,
+        maximum_iterations
+    ):
+        """
+        Check if the optimization parameters are valid for MacroModel.
+
+        MacroModel ``.com`` files have well-defined column widths and
+        special ways to implement too long input. This method raises
+        `MacroModelInputError` exception if the input cannot be
+        converted into a valid ``.com`` file entry.
+
+        Parameters
+        ----------
+        temperature : :class:`float`, optional
+            The temperature in Kelvin at which the MD is run.
+
+        conformers' : :class:`int`, optional
+            The number of conformers sampled and optimized from the MD.
+
+        simulation_time : :class:`float`, optional
+            The simulation time in ``ps`` of the MD.
+
+        time_step : :class:`float`, optional
+            The time step in ``fs`` for the MD.
+
+        eq_time : :class:`float`, optional
+            The equilibriation time in ``ps`` before the MD is run.
+
+        minimum_gradient : :class:`float`, optional
+            The gradient at which optimization is stopped.
+
+        maximum_iterations : :class:`int`, optional
+            The maximum number of iterations done during the
+            optimization.
+
+        Returns
+        -------
+        None : :class:`NoneType`
+
+        """
+
+        # Maximum temperature is `99999.9999`.
+        if temperature > 99999.99:
+            raise MacroModelInputError(
+                'Supplied temperature (> 99999 K) is too high.'
+            )
+
+        # Maximum number of conformers is `9999`.
+        if conformers > 9999:
+            raise MacroModelInputError(
+                'Supplied number of conformers (> 9999) is too high.'
+            )
+
+        # Maximum time is `999999.99`.
+        if simulation_time > 999999.99:
+            raise MacroModelInputError(
+                'Supplied simulation time (> 999999 ps) is too long.'
+            )
+
+        # Maximum time step is `99999.9999`
+        if time_step > 99999.99:
+            raise MacroModelInputError(
+                'Supplied time step (> 99999 fs) is too high.'
+            )
+
+        # Maximum time is `999999.99`.
+        if eq_time > 999999.99:
+            raise MacroModelInputError(
+                'Supplied eq time (> 999999 ps) is too long.'
+            )
+
+        # Minimum convergence gradient is `00000.0001`
```

Comment:
full stop.

Reply:
Also see comment re-redundancy below. and in fact apply this to all the inline comments in this code.